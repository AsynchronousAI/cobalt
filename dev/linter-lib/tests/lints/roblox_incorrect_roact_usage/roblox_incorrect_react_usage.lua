React.createElement("Frame", {
    ThisPropertyDoesntExist = true,
    Size = UDim2.new(1, 0, 1, 0),
    ref = true,
    children = true,
    key = true,

    [React.Event.InputBegan] = function()
    end,

    [React.Event.ThisEventDoesntExist] = function()
    end,
})

local e = React.createElement

e("Frame", {
    Size = UDim2.new(1, 0, 1, 0),
    ThisPropertyDoesntExist = true,
})

e("ThisDoesntExist", {})

e(Components.FooComponent, {
    Foo = 1,
})

call("foo", {})

e("ThisDoesntExist")

e(Components.FooComponent, {
    Name = "Can be passed",
})

React.createElement(Components.FooComponent, {
    Name = "Can be passed",
})

e("Frame", {
    Name = "Should not be passed",
})

e("Frame", {
    Name = "Should.not.be.passed",
})

e("Frame", {
    Name = "0Should0not0be0passed",
})

e("Frame", {
    Name = "Should0not0be0passed",
})

e("Frame", {
    Name = "_Should_not_be_passed",
})

e("Frame", {
    Name = "ShouldNotBePassed",
})

React.createElement("Frame", {
    Name = a,
})
