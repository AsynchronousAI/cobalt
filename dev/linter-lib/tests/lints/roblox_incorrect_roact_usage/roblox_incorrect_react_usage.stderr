error[roblox_incorrect_roact_usage]: `ThisPropertyDoesntExist` is not a property of `Frame`
  ┌─ roblox_incorrect_react_usage.lua:2:5
  │
2 │     ThisPropertyDoesntExist = true,
  │     ^^^^^^^^^^^^^^^^^^^^^^^

error[roblox_incorrect_roact_usage]: `ThisEventDoesntExist` is not a valid event for `Frame`
   ┌─ roblox_incorrect_react_usage.lua:11:5
   │
11 │     [React.Event.ThisEventDoesntExist] = function()
   │     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[roblox_incorrect_roact_usage]: `ThisPropertyDoesntExist` is not a property of `Frame`
   ┌─ roblox_incorrect_react_usage.lua:19:5
   │
19 │     ThisPropertyDoesntExist = true,
   │     ^^^^^^^^^^^^^^^^^^^^^^^

error[roblox_incorrect_roact_usage]: `ThisDoesntExist` is not a valid class
   ┌─ roblox_incorrect_react_usage.lua:22:3
   │
22 │ e("ThisDoesntExist", {})
   │   ^^^^^^^^^^^^^^^^^

error[roblox_incorrect_roact_usage]: `ThisDoesntExist` is not a valid class
   ┌─ roblox_incorrect_react_usage.lua:30:3
   │
30 │ e("ThisDoesntExist")
   │   ^^^^^^^^^^^^^^^^^

error[roblox_incorrect_roact_usage]: `Name` is assigned through the element's key for Roblox instances
   ┌─ roblox_incorrect_react_usage.lua:41:5
   │
41 │     Name = "Should not be passed",
   │     ^^^^
   │
   = try: ["Should not be passed"] = e(...)

error[roblox_incorrect_roact_usage]: `Name` is assigned through the element's key for Roblox instances
   ┌─ roblox_incorrect_react_usage.lua:45:5
   │
45 │     Name = "Should.not.be.passed",
   │     ^^^^
   │
   = try: ["Should.not.be.passed"] = e(...)

error[roblox_incorrect_roact_usage]: `Name` is assigned through the element's key for Roblox instances
   ┌─ roblox_incorrect_react_usage.lua:49:5
   │
49 │     Name = "0Should0not0be0passed",
   │     ^^^^
   │
   = try: ["0Should0not0be0passed"] = e(...)

error[roblox_incorrect_roact_usage]: `Name` is assigned through the element's key for Roblox instances
   ┌─ roblox_incorrect_react_usage.lua:53:5
   │
53 │     Name = "Should0not0be0passed",
   │     ^^^^
   │
   = try: Should0not0be0passed = e(...)

error[roblox_incorrect_roact_usage]: `Name` is assigned through the element's key for Roblox instances
   ┌─ roblox_incorrect_react_usage.lua:57:5
   │
57 │     Name = "_Should_not_be_passed",
   │     ^^^^
   │
   = try: _Should_not_be_passed = e(...)

error[roblox_incorrect_roact_usage]: `Name` is assigned through the element's key for Roblox instances
   ┌─ roblox_incorrect_react_usage.lua:61:5
   │
61 │     Name = "ShouldNotBePassed",
   │     ^^^^
   │
   = try: ShouldNotBePassed = e(...)

error[roblox_incorrect_roact_usage]: `Name` is assigned through the element's key for Roblox instances
   ┌─ roblox_incorrect_react_usage.lua:65:5
   │
65 │     Name = a,
   │     ^^^^
   │
   = try: [a] = React.createElement(...)

